cylc/flow/scripts/cylc_message.py

```diff
@@ -132,6 +134,10 @@ def main(parser, options, *args):
         if message_str == '-':
             pass
         elif ':' in message_str:
+            valid, err_msg = TaskMessageValidator.validate(message_str)
+            if not valid:
+                raise UserInputError(
+                    f'Invalid task message "{message_str}" - {err_msg}')
```

Comment:
The trouble with this is, it doesn't fail until the `cylc message ...` command inside `[runtime][<task>]script` is finally run. Not sure if there is a better way; trying to parse the contents of the `script` setting seems overly complicated

Reply:
Ooh, yeah, hadn't thought of that.  If messages were separate config items, that would be easier, but that train left the station long ago.

Reply:
Actually, maybe this is OK. It results in job, not scheduler, failure. So it can be fixed on the fly (fix task def, reload and retrigger).

Reply:
On that basis I'm OK with this; but we'll see what @oliver-sanders thinks.